### ISSUE TITLE
nk_tooltip() leaks memory

### ISSUE THREAD

**Issue by nsawa** [vurtun/nuklear#287]
I noticed that nk_tooltip() leaks memory. The way to reproduce the problem is as follows:

Change the demo source code a bit so that we can monitor the status of memory allocation.
![1_nuklear_d3d11_h](https://cloud.githubusercontent.com/assets/18512546/20387413/41efe0e6-ad03-11e6-94bf-79abdcb64137.png)
Run the demo and place the mouse cursor on "Hover me for tooltip".
![2_overview](https://cloud.githubusercontent.com/assets/18512546/20387415/47db04cc-ad03-11e6-9f7c-f9c371a2b975.png)
Check out the debug message on the console. Memory allocating is continuously occured approximately every 0.5 seconds. And memory freeing is not occured.
![3_debug_message](https://cloud.githubusercontent.com/assets/18512546/20387418/4c6362fa-ad03-11e6-843c-c59732a2be8f.png)

I think the cause is as follows. At the end of nk_tooltip(), nk_tooltip_end() is called. nk_tooltip_end() calles nk_popup_close() and nk_popup_end(). nk_end() which called from nk_popup_end() has "nk_free_panel()", but it is never processed. Because nk_popup_close() sets the NK_WINDOW_HIDDEN flag. When NK_WINDOW_HIDDEN is set, nk_end() doesn't call nk_free_panel().

I guessed the cause as above, but I have not found a solution. Currently, the only solution I figured out is to avoid using nk_tooltip().

Thanks.

**Comment by nsawa** [comment]
I add a postscript.

I noticed that this is not a problem unique to nk_tooltip(). Memory leak seems to occur also in the following cases:

- Menu
  ```
  nk_menubar_begin(ctx);
  nk_layout_row_static(ctx, 25, 45, 1);
  if (nk_menu_begin_label(ctx, "MENU", NK_TEXT_LEFT, nk_vec2(120, 200))) {
      nk_layout_row_dynamic(ctx, 25, 1);
      if (nk_menu_item_label(ctx, "About", NK_TEXT_LEFT))
          show_app_about = nk_true;
      nk_menu_end(ctx);
  }
  nk_menubar_end(ctx);
  ```
  Click "About" item and then memory leaked.

- Contextual
  ```
  if (nk_contextual_begin(ctx, 0, nk_vec2(100, 300), bounds)) {
      nk_layout_row_dynamic(ctx, 25, 1);
      if (nk_contextual_item_label(ctx, "About", NK_TEXT_CENTERED))
          show_app_about = nk_true;
      nk_contextual_end(ctx);
  }
  ```
  Click "About" item and then memory leaked.

- Combobox
  ```
  nk_layout_row_static(ctx, 25, 200, 1);
  current_weapon = nk_combo(ctx, weapons, LEN(weapons), current_weapon, 25, nk_vec2(200,200));
  ```
  Click any item in the combobox and then memory leaked.

**Comment by vurtun** [comment]
Thanks for the issue, I will fix it as soon as possible.

**Comment by vurtun** [comment]
I pushed a fix in commit e229856391be87957b3d3275351157ff218d0203. Hopefully this is now fixed. I tested it and on my side and memory allocation seems to be correct again.
